libs/platform/src/lib/components/form/textarea/textarea.component.ts

```diff
@@ -0,0 +1,354 @@
+/**
+ * @license
+ * F. Kolar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Input,
+    Optional,
+    Self,
+    ViewChild,
+    ViewEncapsulation,
+    HostListener
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { BaseInput } from '../base.input';
+import { FormFieldControl } from '../form-control';
+
+const VALID_WRAP_TYPES = ['hard', 'soft', 'off'];
+
+export type WrapType = 'hard' | 'soft' | 'off';
+export type FormStates = 'success' | 'error' | 'warning' | 'information' | '';
+
+let uniqueId = 0;
+
+/**
+ * Textarea field implementation to be compliant with our FormGroup/FormField design
+ *
+ */
+@Component({
+    selector: 'fdp-textarea',
+    templateUrl: 'textarea.component.html',
+    styleUrls: ['textarea.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [
+        {
+            provide: FormFieldControl,
+            useExisting: TextareaComponent,
+            multi: true
+        }
+    ]
+})
+export class TextareaComponent extends BaseInput {
+    @Input()
+    height: string;
+
+    @Input()
+    growingMaxLines: number; // rows
+
+    @Input()
+    cols: number;
+
+    @Input()
+    wrap: WrapType = 'soft';
+
+    @Input()
+    readonly: boolean;
+
+    @Input()
+    maxLength: number;
+
+    @Input()
+    showExceededText: boolean = false;
+
+    @Input()
+    growing: boolean = false;
+
+    @Input()
+    textExceededMessage: string;
+
+    /**
+     *  The state of the form control - applies css classes.
+     *  Can be `success`, `error`, `warning`, `information` or blank for default.
+     */
+    @Input()
+    stateType: FormStates = 'warning';
```

Comment:
this shoudl be in the base component

Reply:
fiori design and interaction page for textarea allows application user to specify what state the textarea can fall into once it crosses the max length, for eg, `warning`, `error` etc. 
By default, the stateType is `warning`, but if set to `error`, then beyond the max length character, the state type changes to `error`. 
The base component currently supports only `error` state, and I did not want to change that code for this specific usecase.

Reply:
have removed FormStates and used Status instead, but I still need to take this as Input() because of the design requirement as mentioned in the previous comment.